swiftsimio/writer.py

```diff
@@ -200,6 +200,120 @@ def write_particle_group(self, file_handle: h5py.File, compress: bool):
 
         return
 
+    def write_particle_group_metadata(
+        self, file_handle: h5py.File, dset_attributes: dict
+    ):
+        """
+        Writes the relevant metadata for a particle group
+
+        Parameters
+        ----------
+        file_handle: h5py.File
+            hdf5 output file being written
+        dset_attributes: dict
+            dictionary containg metadata to attach to group
+        """
+        for name, output_handle in getattr(
+            metadata.required_fields, self.particle_name
+        ).items():
+            obj = file_handle[f"/PartType{self.particle_type}/{output_handle}"]
+            for attr_name, attr_value in dset_attributes[output_handle].items():
+                obj.attrs.create(attr_name, attr_value)
+
+        return
+
+    def get_attributes(self, scale_factor: float) -> dict:
+        """
+        Returns a dictionary containg the attributes to attach to the dataset
+
+        Parameters
+        ----------
+        scale_factor: float
+            the cosmological scale factor of the dataset
+
+        Returns
+        -------
+        attributes_dict: dict
+            dictionary containg the attributes applying to the dataset
+        """
+        attributes_dict = {}
+        a_exp_dict = {
```

Comment:
I would prefer it if we could move this to the `metadata`. We already have https://github.com/SWIFTSIM/swiftsimio/blob/master/swiftsimio/metadata/cosmology/cosmology_fields.py, maybe you can figure out how to integrate that here? I don't think we use `cosmology_fields.py` any more, so it may be worth just editing that to make it easier for you.

Reply:
done.